### DISCUSSION TITLE
error en con print(f'') en modulo 8

### DISCUSSION THREAD

**Discussion by NatanDelgado** [LaunchX-InnovaccionVirtual/CursoIntroPython#285] [Q&A] [answered]
buenas a todos

esto repasando los ejemplos del modulo 8 pero al poner el comando 

""""print(f'{planet['name']} polar diameter: {planet['diameter (km)']['polar']}')""""

me da unos errores de que no se cierran ni los parentesis ni los corchetes cuadrados y sale este error 

![image](https://user-images.githubusercontent.com/88174320/154330306-8fba50e3-960b-4fd1-90bd-77d39d3fa7e7.png)

""Unterminated expression in f-string; missing close bracePylance""

alguien sabra porque pasa esto??

![image](https://user-images.githubusercontent.com/88174320/154329652-31890e25-fb3b-4947-aad6-35163024abf4.png)

**Comment by kriztobal** [comment, marked as answer]
Puedes usar el f con comillas dobles en las orillas, para que no se confunda con las comillas sencillas que están dentro

**Reply by NatanDelgado** [reply]
si correcto justo acabo de ver ese error, muchas gracias ya quedo

**Comment by EnriqueIbarraM** [comment]
Dentro de los corchetes [ ] utiliza comillas dobles, ya que tu cadena la estás "definiendo" con comilla simple, entonces cuando colocas la primer comilla simple dentro de los corchetes, lo interpreta como el fin de la cadena a imprimir. Por ello no encuentra el corchete de cierre.

**Reply by NatanDelgado** [reply]
si correcto justo acabo de ver ese error, puse mejor dobles comillas en la cadena "f" muchas gracias ya quedo

**Comment by MarcosSilvaG** [comment]
el error está en el escapado de datos, aunque parezca lo mismo poner dobles comillas que comillas simples estas funcionan diferente dentro del código.
Una manera fácil de verlo es utilizar comillas dobles para todos tus prints y en caso de que dentro de este print necesites agregar información que lleve comillas utilices las simples para que python sepa interpretar las instrucciones.
```py
print(f"{planet['name'] "}
```

**Reply by NatanDelgado** [reply]
justo asi lo arregle muchas gracias, ya quedo

**Comment by ErikGustavoAvendanoG** [comment]
Solo como sugerencia y que me ha servido de mucho, Visual Studio te marca de un color diferente cada cosa que vas escribiendo, de un color las variables, de otro las cadenas. Entonces creo que igual eso te podría ayudar a mi me ha servido bastante.
Ahí por ejemplo te marca de un color el nombre de tu variable al principio del print pero despues te marca de otro color (del mismo color el nombre de una variable que el corchete que abre)
